app/assets/javascripts/checklist-utils.js

```diff
@@ -0,0 +1,52 @@
+// Utility functions used in checklist views (UserChecklist, AdminOnly::MasterChecklist
+
+"use strict";
+
+const defaultHideClass = "hide";
+const defaultShowClass = "inherit";
+
+
+function showElementWithClass(element, showClass = defaultShowClass, hideClass = defaultHideClass) {
+  replaceClass(element, hideClass, showClass);
+}
```

Comment:
Since we're using bootstrap shouldn't we be using bootstrap classes (here and throughout)?  e.g. _d-none_ instead of _hide_

Reply:
Good suggestion. :-)  I'll look into it.

Reply:
Maybe change call to `replaceClass` with this:
```
$(element).toggleClass(showClass, true).toggleClass(hideClass, false);
```
Similarly, the `true` and `false` values would be swapped for `hideElementWithClass`.

https://api.jquery.com/toggleclass/

Reply:
@patmbolger - This is ready for another review.
 I've updated to use jQuery:  I deleted the `checklist-utils.js` file, and so all of the javascript is currently in `show_progress.html.haml`.  Once the javascript in this file is OK, then in later PRs as I actually pull the pieces together to implement all of this, this javascript might be used on other pages and so might move into `assets/javascripts/...`

Right now I want to make sure that the javascript is good, no matter what file it is in. 


Specific js-related changes:
- got rid of all the tiny functions to get an element Id
- now use `Utility.checkHttpError` to check for errors returned from ajax call (I copied the updated and renamed function from your subcategories branch.)
   - updated `handle_xhr_request` in ApplicationController so that the keys it uses for the response status are consistent with what `Utility.checkHttpError` expects. (I renamed `status_text` to `statusText` etc.
- using the Bootstrap `d-none` class to hide elements


Also:
- changed the CSS id `#no-longer-in-use` to a class